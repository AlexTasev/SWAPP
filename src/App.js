import React, { useState } from 'react';
import './App.css';
import ApolloClient from "apollo-client";
import { ApolloProvider } from "react-apollo";
import { BrowserRouter } from 'react-router-dom';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { createHttpLink } from 'apollo-link-http';
import Home from './Home';
import authLink from './client/auth';
import {typeDefs} from './client/local'
import { ThemeProvider } from 'styled-components';
import { lightTheme, darkTheme } from './theme';
import { GlobalStyles } from './global';

const cache = new InMemoryCache();

const httpLink = createHttpLink({
  uri: 'http://softuni-swapp-212366186.eu-west-1.elb.amazonaws.com/graphql',
});

const client = new ApolloClient({
  cache,
  link: authLink.concat(httpLink),
  typeDefs
});

const isAuthed = !!localStorage.getItem('token');


cache.writeData({
  data: {
    authenticated: isAuthed,
  },
});


function App() {

  const [theme, setTheme] = useState('light');
  const toggleTheme = () => {
    if (theme === 'light') {
      setTheme('dark');
    } else {
      setTheme('light');
    }
  }

  console.log(toggleTheme);
  
  return (
    <ApolloProvider client={client}>
      <BrowserRouter>
      <div className="App">
      <ThemeProvider theme={theme === 'light' ? lightTheme : darkTheme}>
      <GlobalStyles />
        <Home toggleTheme = {{toggleTheme}}/>
      </ThemeProvider>
      </div>
      </BrowserRouter>
    </ApolloProvider>
  );
}

export default App;