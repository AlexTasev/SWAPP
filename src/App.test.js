import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { ApolloProvider } from "react-apollo";
import { client } from './App'
const ReactTestRenderer = require('react-test-renderer');

describe('test Episodes component', () => {
  //Check if the component renders
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  // Snapshot test to make sure the UI does not change unexpectedly
  it('Should compare the component with a snapshot', () => {
    const component = ReactTestRenderer.create(
      <ApolloProvider client={client}>
        <App />
      </ApolloProvider>
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  })

});