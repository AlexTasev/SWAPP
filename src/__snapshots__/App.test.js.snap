// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Episodes component Should compare the component with a snapshot 1`] = `
<div
  className="App"
>
  <div
    className="site-wrapper"
  >
    <div
      className="container"
    >
      <section
        className="login-form"
      >
        <h1
          className="logo"
          onClick={[Function]}
        >
          SWAPP
        </h1>
        <form
          className="css-1rta19r"
          onSubmit={[Function]}
        >
          <div
            className="css-168ufxn"
          >
            <div
              className="css-uq1pv2"
            >
              <input
                className="css-1urscag"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
              />
              <input
                className="css-4y926w"
                onChange={[Function]}
                placeholder="Enter your password"
                type="password"
                value=""
              />
            </div>
            <div
              className="css-lcap0w"
            >
              <button
                className="css-vhv7r1"
                type="submit"
              >
                Login
              </button>
            </div>
          </div>
        </form>
      </section>
    </div>
  </div>
</div>
`;
